src/Http/Http.Extensions/gen/StaticRouteHandlerModel/EndpointResponse.cs

```diff
@@ -27,18 +27,20 @@ internal class EndpointResponse
     internal EndpointResponse(IMethodSymbol method, WellKnownTypes wellKnownTypes)
     {
         WellKnownTypes = wellKnownTypes;
-        ResponseType = UnwrapResponseType(method);
+        ResponseType = UnwrapResponseType(method, out bool isAwaitable, out bool awaitableIsVoid);
         WrappedResponseType = method.ReturnType.ToDisplayString(EmitterConstants.DisplayFormat);
-        IsAwaitable = GetIsAwaitable(method);
-        IsVoid = method.ReturnsVoid;
+        IsAwaitable = isAwaitable;
+        IsVoid = method.ReturnsVoid || awaitableIsVoid;
         IsIResult = GetIsIResult();
         IsSerializable = GetIsSerializable();
         ContentType = GetContentType(method);
         IsAnonymousType = method.ReturnType.IsAnonymousType;
     }
 
-    private ITypeSymbol? UnwrapResponseType(IMethodSymbol method)
+    private ITypeSymbol? UnwrapResponseType(IMethodSymbol method, out bool isAwaitable, out bool awaitableIsVoid)
```

Comment:
I wonder if we are going to use this logic multiple times. Perhaps we could move this to `SymbolExtensions.cs` along with other Roslyn symbol related helpers.

Reply:
TBH, I think the previous iteration of this `GetIsAwaitable` might've been a better candidate for code-sharing. My main hesitancy about making this shared code is it assumes that the caller  is able to provide access to a well-known types cache with the appropriate symbols loaded.